source/systems/common.rst

```diff
@@ -64,5 +64,1789 @@ RDHPCS Platforms
+
+
+Setting the --cpus-per-task parameter
+"""""""""""""""""""""""""""""""""""""
+
+.. note::
+
+   We are still investigating this issue and will update it as we learn more.
+
+
+With the update of Slurm from 20.11.7 to 21.08.5 , the behavior of
+enforcing memory limit for the job-step by Slurm has changed from the previous
+version resulting in some users seeing **"Out of Memory"** or **OOM**
+errors. Please note that this is not a bug but a fix for an issue that was
+more lax in its previous implementation.  Jobs that took advantage of the lax
+implementation in the previous version may see the "out-of-memory" errors with
+this new implementation. <p> Please keep in mind that when you specify
+**"--cpus-per-task"** option on the sbatch command, **each job-step only
+gets its proportional share of memory!**  If you leave that option out on the
+sbatch command, or if you specify **"--mem=0"** then each of the job-steps,
+launched by srun within a job, will be able to use all the available memory
+allocated for that job.  Please note that how much memory is allcated to the
+job depends on whether the job is a "shared" job or an "exclusive" job.  On
+Hera and Jet, currently all the non-serial jobs get exclusive access to the
+node whereas on the MSU systems the default is "non-exclusive".  It is worth
+noting that as core counts increase on the newer processors "exclusive" will be
+the preferred option, so non-MSU systems also may move towards "non-exclusive"
+or "shared" mode in the future.
+
+
+There may be situations your job where you do need this option on sbatch for
+your parallel job-step but there are serial job-steps in the same job that need
+more memory than what is available based on the above, in which case it can
+over ridden on the `srun` command.
+
+Here is a concrete example:
+
+.. code-block:: shell
+
+   #!/bin/bash -l
+   #SBATCH ...
+   #SBATCH --cpus-per-task=2           # Each job-step by default only gets 2 CPUs worth of memory
+   #SBATCH
+
+   srun $wrf_exe                    # 2 CPUs worth of memory is sufficient for this task
+
+   #  The next task requires 4 times each CPU's share of memory, so we can override it thus
+   # even the post processor is a serial task
+
+   srun --cpus-per-task=4 $post        # Without that it will only 2 CPUs worth of memory
+
+
+As mentioned above, an alternative solution is specify **"--mem=0"** as one of
+the SBATCH directives in the job or as a command line option instead of
+modifying the srun line.
+
+Specifying a processor layout for your job (non-uniform or arbitrary example)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Some applications require tasks to be laid out non-uniformly. For example a
+number of applications have task0 requiring more memory than the remaining
+tasks. This requires the capability of doing non-uniform task layout.
+
+The script named ``arbitrary.pl`` is available when module
```

Comment:
The above section mentioned this `arbitrary.pl` but does not explain how to get this script.

More information is needed.

Reply:
removed